### PR TITLE
Add optional support for json column data storage

### PR DESCRIPTION
**Pull request by twelve17**
This is a follow up to #91 with a draft of how the unit tests could be implemented to cover the JSON column changes introduced in that PR.

@chill117 - Thanks so much for writing this library 🎉 

Let me know what you think of how I am adding coverage of the `jsonData` flag.  I went with a dynamic test approach since I thought it would be good to cover the touch points broadly, though it may have caused testing in places where the flag doesn't matter.   I thought that might reduce repetition/verbosity in the test code, but let me know if you'd prefer to structure it differently and I'll update this PR.

Regarding your point about the two schema files: I too am torn about this. I don't like the repetition, but wrapping the SQL in some generator also seems a bit overly complex.  I am leaning towards "keep the two files and wait for a bigger change in the future that warrants the added complexity", but I could probably go either way, and if you want, I can experiment with a solution where the sql is dynamically generated.

### CODE REVIEW

File: test/unit/get.js [hunk 1]
```diff
@@ -7,60 +7,76 @@ var manager = require('../manager');
...
+				});
+			})
+
+			describe('when a session does not exist', function() {
 
-		beforeEach(manager.clearSessions);
+				beforeEach(manager.clearSessions);
 
-		it('should return NULL', function(done) {
+				it('should return NULL', function(done) {
```
Comment (twelve17) [inline test/unit/get.js:68]: I am not sure why GH is highlighting this here. I did try to preserve tab formatting in the file (checked with the linter).  😅

File: test/manager.js [hunk 2]
```diff
@@ -3,7 +3,6 @@
 var _ = require('underscore');
 var async = require('async');
 var session = require('express-session');
-var mysql = require('mysql');
```
Comment (twelve17) [inline test/manager.js:6]: This removal was carried over from #91 . I didn't know if this require was needed here either?

**Comment by twelve17** [comment]
Whoops, I just noticed [in this comment](https://github.com/chill117/express-mysql-session/issues/91#issuecomment-582855571) that the `constructor` test should also be updated. I'l add a test now.

File: test/unit/constructor.js [hunk 3]
```diff
@@ -169,6 +169,37 @@ describe('constructor', function() {
...
+				var options = _.extend({}, manager.config);
+
+				sessionStore = new MySQLStore(options);
+				done = _.once(done);
+
+				expect(sessionStore.options.jsonData).to.be.undefined;
+
+				setTimeout(function() {
+					done();
+				}, 30);
```
Comment (twelve17) [inline test/unit/constructor.js:185]: I duplicated the pattern of the custom `done()` function and `setTimeout`, as these new tests were hanging the test suite. I presume the connection is what is hanging the tests, and there's something in that approach to let the connection get closed?

**Comment by chill117** [comment]
Thanks for taking the time to create a pull-request. I will close this PR because in the end it was not necessary to add a new option. The module already worked with the JSON type for the data column. I've added tests for this specific case and a clarified the readme to help anyone else who might want to use the JSON type - see [Custom database table schema](https://github.com/chill117/express-mysql-session#custom-database-table-schema)

**Pull request closed without merging**